supervise.go

```diff
@@ -107,7 +113,16 @@ func (lrb *lineRingBuffer) Write(b []byte) (int, error) {
 			break
 		}
 
-		lrb.r.Value = text[:idx]
+		line := text[:idx]
+		lrb.r.Value = line
+		for stream := range lrb.streams {
```

Comment:
You need to hold the mutex when accessing `lrb.streams`, no?

When adding and deleting, you’re using the mutex correctly.

Reply:
The lock is already taken in line 107 (at the beginning of `Write`)